### PR TITLE
fix issues:Image file not deleted when a doc is removed #9541

### LINKED ISSUE #9541: Image file not deleted when a doc is removed
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.9.2

### Cloud or Self Hosted

Self Hosted (Source)

### Steps to reproduce

1. add a docx file containing images in a knowledgebase
2. remove the above file
3. view the folder "image_files" and "upload_files"

### ✔️ Expected Behavior

Image files in the docx file should be deleted

### ❌ Actual Behavior

Image files in the docx file not deleted

### PR DESCRIPTION
**Pull request by liuhaoran1212**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] Please open an issue before creating a PR or link to an existing issue
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Fixed the following issues: When deleting a doc in the web, the doc in "uplaod_files" folder is deleted, but image files extrating from the doc remains in "image_files" folder

Fixes #9541 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: api/core/tools/utils/web_reader_tool.py [hunk 1]
```diff
@@ -356,3 +356,16 @@ def content_digest(element):
                 digest.update(child.encode("utf-8"))
             digest = digest.hexdigest()
     return digest
+
+
+def get_image_upload_file_ids(content):
+    pattern = r"!\[image\]\((.*?file-preview)\)"
```
Comment (laipz8200) [inline api/core/tools/utils/web_reader_tool.py]: You need to also find the pattern contains `image-preview`

**Review by laipz8200 — DISMISSED** [review]
LGTM

File: api/core/tools/utils/web_reader_tool.py (hunk 1 above)
Comment (liuhaoran1212) [reply api/core/tools/utils/web_reader_tool.py]: Okay, I have modified code and made a commit, thanks.

**Review by laipz8200 — APPROVED** [review]
LGTM

**Pull request merged**